### PR TITLE
chore: Prepare 41.0.0-rc1

### PR DESCRIPTION
**Pull request by andygrove**
## Which issue does this PR close?

Part of https://github.com/apache/datafusion/issues/11476

## What changes are included in this PR?

- Bump version
- New [changelog](https://github.com/andygrove/datafusion/blob/prep-41-rc1/dev/changelog/41.0.0.md)

### CODE REVIEW

File: dev/changelog/41.0.0.md [hunk 1]
```diff
@@ -0,0 +1,363 @@
...
+KIND, either express or implied.  See the License for the
+specific language governing permissions and limitations
+under the License.
+-->
+
+# Apache DataFusion 41.0.0 Changelog
+
+This release consists of 245 commits from 69 contributors. See credits at the end of this changelog for more information.
+
+**Breaking changes:**
```
Comment (alamb) [inline dev/changelog/41.0.0.md:24]: this is a pretty sweet list of changes

File: dev/changelog/41.0.0.md [hunk 2]
```diff
@@ -0,0 +1,363 @@
...
+- Improve nested loop join code [#11863](https://github.com/apache/datafusion/pull/11863) (lewiszlw)
+- [Minor]: Refactor to use Result.transpose() [#11882](https://github.com/apache/datafusion/pull/11882) (djanderson)
+- support `ANY()` op [#11849](https://github.com/apache/datafusion/pull/11849) (samuelcolvin)
+
+## Credits
+
+Thank you to everyone who contributed to this release. Here is a breakdown of commits (PRs merged) per contributor.
+
+```
+    48	Andrew Lamb
```
Comment (alamb) [inline dev/changelog/41.0.0.md:292]: admittedly I stack the deck here with many small PRs

**Review by alamb — COMMENTED** [review]
Thank you @andygrove

**Review by viirya — APPROVED** [review]

**Comment by viirya** [comment]
Not sure about the CI failure:
```
Binary files a/datafusion/core/example.parquet and b/datafusion/core/example.parquet differ
```

Re-triggered to verify it.

**Comment by andygrove** [comment]
> Not sure about the CI failure:
> 
> ```
> Binary files a/datafusion/core/example.parquet and b/datafusion/core/example.parquet differ
> ```
> 
> Re-triggered to verify it.

The `example.parquet` file will change every time we increase the version number, because we store metadata in the parquet file witht the version of DataFusion that was used when writing. This is not ideal... for now I will regenerate this file as part of this PR.

**Comment by alamb** [comment]
> > Not sure about the CI failure:
> > ```
> > Binary files a/datafusion/core/example.parquet and b/datafusion/core/example.parquet differ
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > Re-triggered to verify it.
> 
> The `example.parquet` file will change every time we increase the version number, because we store metadata in the parquet file witht the version of DataFusion that was used when writing. This is not ideal... for now I will regenerate this file as part of this PR.

**Comment by alamb** [comment]
Sorry -- I clicked the wrong button and closed this PR by accident

**Comment by alamb** [comment]
> > Not sure about the CI failure:
> > ```
> > Binary files a/datafusion/core/example.parquet and b/datafusion/core/example.parquet differ
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > Re-triggered to verify it.
> 
> The `example.parquet` file will change every time we increase the version number, because we store metadata in the parquet file witht the version of DataFusion that was used when writing. This is not ideal... for now I will regenerate this file as part of this PR.

I will file a ticket

**Comment by alamb** [comment]
Filed https://github.com/apache/datafusion/issues/11892 to track

**Review by alamb — APPROVED** [review]

**Pull request merged**